abd-clam/src/core/graph/_graph.rs

```diff
@@ -142,60 +164,66 @@ impl<'a, T: Number, U: Number, D: Dataset<T, U>> Graph<'a, T, U, D> {
-        assert!(!clusters.is_empty());
-
-        let (population, min_depth, max_depth) =
-            clusters
-                .iter()
-                .fold((0, usize::MAX, 0), |(population, min_depth, max_depth), &c| {
-                    (
-                        population + c.cardinality,
-                        std::cmp::min(min_depth, c.depth()),
-                        std::cmp::max(max_depth, c.depth()),
-                    )
-                });
-
-        let adjacency_map = {
-            let mut adjacency_map: AdjacencyMap<T, U, D> = clusters.iter().map(|&c| (c, HashSet::new())).collect();
-            edges.iter().for_each(|&e| {
-                adjacency_map.get_mut(e.left()).unwrap().insert(e.right());
-                adjacency_map.get_mut(e.right()).unwrap().insert(e.left());
-            });
-            adjacency_map
-        };
-
-        Self {
-            ordered_clusters: clusters.iter().copied().collect(),
-            clusters,
-            edges,
-            adjacency_map,
-            population,
-            min_depth,
-            max_depth,
-            distance_matrix: None,
-            adjacency_matrix: None,
-            frontier_sizes: None,
-        }
-    }
-
-    #[allow(dead_code)]
+    #[allow(unused_mut, unused_variables, dead_code)]
+    #[allow(clippy::needless_pass_by_value)]
+    pub fn new(_clusters: ClusterSet<'a, U>, _edges: EdgeSet<'a, U>) -> Self {
+        todo!()
+
+        // assert!(!clusters.is_empty());
+        //
+        // let (population, min_depth, max_depth) =
+        //     clusters
+        //         .iter()
+        //         .fold((0, usize::MAX, 0), |(population, min_depth, max_depth), &c| {
+        //             (
+        //                 population + c.cardinality(),
+        //                 std::cmp::min(min_depth, c.depth()),
+        //                 std::cmp::max(max_depth, c.depth()),
+        //             )
+        //         });
+        //
+        // let adjacency_map = {
+        //     let mut adjacency_map: AdjacencyMap<U> = clusters.iter().map(|&c| (c, HashSet::new())).collect();
+        //     edges.iter().for_each(|&e| {
+        //         adjacency_map.get_mut(e.left()).unwrap().insert(e.right());
+        //         adjacency_map.get_mut(e.right()).unwrap().insert(e.left());
+        //     });
+        //     adjacency_map
+        // };
+        //
+        // Self {
+        //     ordered_clusters: clusters.iter().copied().collect(),
+        //     clusters,
+        //     edges,
+        //     adjacency_map,
+        //     population,
+        //     min_depth,
+        //     max_depth,
+        //     distance_matrix: None,
+        //     adjacency_matrix: None,
+        //     frontier_sizes: None,
+        // }
+    }
+
+    #[allow(clippy::missing_docs_in_private_items)]
+    #[allow(unused_mut, unused_variables, dead_code)]
     fn compute_distance_matrix(&self) -> Vec<Vec<U>> {
```

Comment:
Add docs here and remove the missing docs macro. This function should create a square, symmetric matrix of edge lengths, leaving zero distances for when there is no edge